api/src/main/java/org/openmrs/module/fhir/api/util/FHIRObsUtil.java

```diff
@@ -240,4 +241,51 @@ public static Observation generateObs(Obs obs) {
 		}
 		return observation;
 	}
+	
+	public static Obs generateOpenMRSObs(Observation observation, List<String> errors) {
+		Obs obs = new Obs();
+		
+		obs.setComment(observation.getComments());
+		if (observation.getSubject() != null) {
+			ResourceReferenceDt subjectref = observation.getSubject();
+			IdDt id = subjectref.getReference();
+			String uri = id.getValue();
+			String[] paths = uri.split("/");
```

Comment:
You should be able to use subjectref.getIdPart(). Please fix this and will merge the pull